pkg/crd/testdata/cronjob_types.go

```diff
@@ -120,6 +120,18 @@ type CronJobSpec struct {
 	// +kubebuilder:example={{nested: {foo: "baz", bar: true}},{nested: {bar: false}}}
 	DefaultedObject []RootObject `json:"defaultedObject"`
 
+	// This tests that empty slice defaulting can be performed.
+	// +kubebuilder:default={}
```

Comment:
Is `{}` a valid default for slice?

Reply:
Good question, this needs to be fixed. CRD does not complain for deafult: [] though

Reply:
Everything is fixed in the current commit.

Reply:
This is strange. But I guess fine given it's not JSON but our custom syntax (?)

Reply:
I assume/hope we're parsing `{1,2}` correctly as an array?

Reply:
I agree it's strange, but too late to change it now? Probably a reason for not using the obvious `[]` for arrays in marker syntax?

Reply:
Yup seems definitely too late to change it now. Given that non-empty arrays worked already in the past :)